sympy/printing/theanocode.py

```diff
@@ -134,7 +134,7 @@ def _print_MatMul(self, expr, **kwargs):
     def _print_MatPow(self, expr, **kwargs):
         children = [self._print(arg, **kwargs) for arg in expr.args]
         result = 1
-        if isinstance(children[1], int):
+        if isinstance(children[1], int) and children[1] > 0:
```

Comment:
This should error if the power isn't a positive integer. Right now it will return 1

Reply:
Fair enough, will raise a `NotImplementedError`